cognify/optimizer/core/opt_layer.py

```diff
@@ -801,13 +802,83 @@ def _optimize(self):
                     log_id, result = f.result()
                     if result and result.complete:
                         self._save_opt_ckpt()
+                        
+    def _optimize_SH(self):
+        opt_config = self.top_down_info.opt_config
+        
+        bucket_size = opt_config.throughput
```

Comment:
i don't think users know how to set the bucket_size, and it's not so sensitive to different workflows/budgets, right? then we can just make it an internal configuration not exposed to user.